examples/vision/randaugment.py

```diff
@@ -0,0 +1,296 @@
+transforms like color jitters, Gaussian blurs, saturations, etc. along with more
+traditional augmentation transforms such as random crops.
+
+RandAugment has two parameters:
+* `n` that denotes the number of randomly selected augmentation transforms to apply
+sequentially
+* `m` strength of all the augmentation transforms
+
+These parameters are tuned for a given dataset and a network architecture. The authors of
+RandAugment also provide pseudocode of RandAugment in the original paper (Figure 2):
+
+![](https://i.ibb.co/Df6Ynxd/image.png)
+
+Recently, it has been a key component of works like [Noisy Student Training](https://arxiv.org/abs/1911.04252) and
+[Unsupervised Data Augmentation for Consistency Training](https://arxiv.org/abs/1904.12848). It has been also central to the
+success of [EfficientNets](https://arxiv.org/abs/1905.11946).
+
+This example requires TensorFlow 2.4 or higher, as well as
+[`imgaug`](https://imgaug.readthedocs.io/), which can be installed using the following
+command:
+"""
+
+"""shell
+pip install -U -q imgaug
+"""
+
+"""
+## Setup
+"""
+import numpy as np
+import tensorflow as tf
+import matplotlib.pyplot as plt
+from tensorflow.keras import layers
+
+import tensorflow_datasets as tfds
+
+tfds.disable_progress_bar()
+
+from imgaug import augmenters as iaa
+import imgaug as ia
+
+"""
+## Load the CIFAR10 dataset
+
+For this example, we will be using the [CIFAR10
+dataset](https://www.cs.toronto.edu/~kriz/cifar.html).
+"""
+
+(x_train, y_train), (x_test, y_test) = tf.keras.datasets.cifar10.load_data()
+print(f"Total training examples: {len(x_train)}")
+print(f"Total test examples: {len(x_test)}")
+
+"""
+## Define hyperparameters
+"""
+
+AUTO = tf.data.AUTOTUNE
+BATCH_SIZE = 128
+EPOCHS = 10
+
+"""
+## Initialize `RandAugment` object
+
+Now, we will initialize a `RandAugment` object from the `imgaug.augmenters` module with
+the parameters suggested by the RandAugment authors.
+"""
+
+rand_aug = iaa.RandAugment(n=3, m=7)
+
+
+def augment(images):
+    # Input to `augment()` is a TensorFlow tensor which
+    # is not supported by `imgaug`. This is why we first
+    # convert it to its `numpy` variant.
+    return rand_aug(images=images.numpy())
+
+
+"""
+## Create TensorFlow `Dataset` objects
+
+There's one problem, though. We cannot map our `augment()` function to a TensorFlow
+`Dataset` object. To tackle this problem, we will make use of the
+[`tf.py_function`](https://www.tensorflow.org/api_docs/python/tf/py_function) that can
+convert a Python function into a TensorFlow op.
+"""
+
+train_ds_rand = (
+    tf.data.Dataset.from_tensor_slices((x_train, y_train))
+    .shuffle(BATCH_SIZE * 100)
+    .batch(BATCH_SIZE)
+    .map(
+        lambda x, y: (tf.py_function(augment, [x], [tf.float32]), y),
+        num_parallel_calls=AUTO,
+    )
+    # The returned output contains an unncessary axis of
+    # 1-D and we need to remove it.
+    .map(lambda x, y: (tf.squeeze(x), y), num_parallel_calls=AUTO)
```

Comment:
Why? Also, seems preferable to put it in the `augment` function (in this case, that would justify not using a lambda)

Reply:
My guess is while defining `py_function` we are defining the output to be a list (`[x]`) which is what I saw in the docs as well. This is what is likely causing this issue.

Reply:
What happens if you do `tf.py_function(augment, x, tf.float32), y)`?

Reply:
Will try out and get back.

Reply:
Works perfectly fine. Just pushed the changes.